File: app/src/main/java/com/rahmad/popularmoviesstage2/util/AppSharedPref.java
package com.rahmad.popularmoviesstage2.util;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by hartviq on 10/10/16.
 * inbox.rahmad@gmail.com
 * Copyright 2016
 */

public class AppSharedPref {

  private static final String FILENAME = "APP_SHARED_DATA";
  private final String sortingState = "SORTING_STATE";

  private final SharedPreferences sharedPreferences;
  private final SharedPreferences.Editor editor;

  @SuppressLint("CommitPrefEdits")
  public AppSharedPref(Context context) {
    sharedPreferences = context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE);
    editor = sharedPreferences.edit();
  }

  public Integer getSortingState() {
    return sharedPreferences.getInt(sortingState, 1);
  }

  public void setSortingState(Integer sortingState) {
    editor.putInt(this.sortingState, sortingState);
    editor.apply();
  }

  @SuppressWarnings("unused")
  public void clear() {
    editor.clear();
    editor.commit();
  }

  @SuppressWarnings("UnusedReturnValue")
  public boolean commit() {
    return editor.commit();
  }
}

File: app/src/main/java/com/rahmad/popularmoviesstage2/util/AppSharedPref.kt
package com.rahmad.popularmoviesstage2.util

import android.content.Context
import android.content.SharedPreferences

class AppSharedPref(context: Context) {

    private val FILENAME = "APP_SHARED_DATA"
    private val sortingState = "SORTING_STATE"

    private val sharedPreferences: SharedPreferences
    private val editor: SharedPreferences.Editor

    init {
        sharedPreferences = context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE)
        editor = sharedPreferences.edit()
    }

    fun getSortingState(): Int {
        return sharedPreferences.getInt(sortingState, 1)
    }

    fun setSortingState(sortingState: Int) {
        editor.putInt(this.sortingState, sortingState)
        editor.apply()
    }

    fun clear() {
        editor.clear()
        editor.apply()
    }

    fun commit(): Boolean {
        return editor.commit()
    }
}
